database/scripts/generate_report.rb

```diff
@@ -25,14 +25,15 @@
 options[:report_name] = "buildfarm-report_#{DateTime.now.strftime("%Y-%m-%d_%H-%M")}.json" if options[:report_name].nil?
 
 def generate_report(report_name, exclude_set)
-    report_regressions_all = BuildfarmToolsLib::test_regressions_all(filter_known: true)
+    report_regressions_all = BuildfarmToolsLib::test_regressions_all(filter_known: true).freeze # Freeze as this is a read-only variable
```

Comment:
What does this freeze do?

Reply:
After some reading I understood it makes the variable read-only, why was it needed to make this work as expected?

Reply:
> why was it needed to make this work as expected?

Because the next line modified the report regressions, and the flaky test regressions method received a different collection of test regressions. Currently, freezing is not necessary, but it will help us don't shoot in the shoe when using the variable, and explicitly declare it as read-only